core/src/test/java/org/jobrunr/server/carbonaware/CarbonIntensityForecastTest.java

```diff
@@ -0,0 +1,243 @@
+
+        try (var ignored = InstantMocker.mockTime("2020-01-01T08:00:00Z")) {
+            assertThat(carbonIntensityForecast.lowestCarbonIntensityInstant(from, to)).isEqualTo(Instant.parse("2020-01-01T08:00:00Z"));
+        }
+    }
+
+    @Test
+    void lowestCarbonIntensityInstantReturnsNullIfNoForecastAvailable() {
+        Instant from = Instant.parse("2020-01-01T00:00:00Z");
+        Instant to = Instant.parse("2020-01-01T10:00:00Z");
+        CarbonIntensityForecast carbonIntensityForecast = new CarbonIntensityForecast(new ApiResponseStatus("OK", ""), "ENTSO-E", "10Y1001A1001A82H", "Germany", "Europe/Berlin", Instant.now().plus(1, DAYS), null);
+
+        assertThat(carbonIntensityForecast.lowestCarbonIntensityInstant(from, to)).isNull();
+    }
+
+    @Test
+    void lowestCarbonIntensityInstantIsBetweenFromAndTo() {
+        Instant from = Instant.parse("2020-01-01T07:00:00Z");
+        Instant to = Instant.parse("2020-01-01T09:00:00Z");
+        ArrayList<TimestampedCarbonIntensityForecast> intensityForecast = new ArrayList<>(Arrays.asList(
+                new TimestampedCarbonIntensityForecast(Instant.parse("2020-01-01T06:00:00Z"), Instant.parse("2020-01-01T07:00:00Z"), 0),
+                new TimestampedCarbonIntensityForecast(Instant.parse("2020-01-01T07:00:00Z"), Instant.parse("2020-01-01T08:00:00Z"), 3),
+                new TimestampedCarbonIntensityForecast(Instant.parse("2020-01-01T08:00:00Z"), Instant.parse("2020-01-01T09:00:00Z"), 2),
+                new TimestampedCarbonIntensityForecast(Instant.parse("2020-01-01T09:00:00Z"), Instant.parse("2020-01-01T10:00:00Z"), 4),
+                new TimestampedCarbonIntensityForecast(Instant.parse("2020-01-01T10:00:00Z"), Instant.parse("2020-01-01T11:00:00Z"), 1)
+        ));
+        CarbonIntensityForecast carbonIntensityForecast = new CarbonIntensityForecast(new ApiResponseStatus("OK", ""), "ENTSO-E", "10Y1001A1001A82H", "Germany", "Europe/Berlin", Instant.now().plus(1, DAYS), intensityForecast);
+
+        try (MockedStatic<Instant> ignored = InstantMocker.mockTime("2020-01-01T00:00:00Z")) {
+            assertThat(carbonIntensityForecast.lowestCarbonIntensityInstant(from, to)).isEqualTo(Instant.parse("2020-01-01T08:00:00Z"));
+        }
+    }
+
+
+    @Test
+    void lowestCarbonIntensityInstantAfterNow() {
+        Instant from = Instant.parse("2020-01-01T00:00:00Z");
+        Instant to = Instant.parse("2020-01-01T23:00:00Z");
+        ArrayList<TimestampedCarbonIntensityForecast> intensityForecast = new ArrayList<>(Arrays.asList(
+                new TimestampedCarbonIntensityForecast(Instant.parse("2020-01-01T08:00:00Z"), Instant.parse("2020-01-01T09:00:00Z"), 0),
+                new TimestampedCarbonIntensityForecast(Instant.parse("2020-01-01T09:00:00Z"), Instant.parse("2020-01-01T10:00:00Z"), 1)
+        ));
```

Comment:
Why are we wrapping Arrays.asList() with a new ArrayList?

Reply:
Probably because the constructor only accepts ArrayList; changed to List.